File: mobile/src/main/java/rs/readahead/washington/mobile/domain/repository/uwazi/ICollectUwaziTemplatesRepository.java
package rs.readahead.washington.mobile.domain.repository.uwazi;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;
import rs.readahead.washington.mobile.domain.entity.uwazi.CollectTemplate;
import rs.readahead.washington.mobile.domain.entity.uwazi.ListTemplateResult;
import rs.readahead.washington.mobile.domain.entity.uwazi.UwaziEntityInstance;

public interface ICollectUwaziTemplatesRepository {
     Single<List<CollectTemplate>> listBlankTemplates();
     Single<List<CollectTemplate>> listFavoriteTemplates();
     Single<ListTemplateResult> updateBlankTemplates(ListTemplateResult listTemplateResult);
     Single<ListTemplateResult> updateBlankTemplatesIfNeeded(ListTemplateResult listTemplateResult);
     Single<CollectTemplate> updateBlankTemplate(CollectTemplate template);
     Single<CollectTemplate> saveBlankTemplate(CollectTemplate template);
     Single<CollectTemplate> getBlankCollectTemplateById(String templateID);
     Single<UwaziEntityInstance> saveEntityInstance(UwaziEntityInstance instance);
     Completable deleteTemplate(final long id);
     Completable deleteEntityInstance(final long id);
     Single<CollectTemplate> toggleFavorite(CollectTemplate template);
     Single<List<UwaziEntityInstance>> listDraftInstances();
     Single<List<UwaziEntityInstance>> listOutboxInstances();
     Single<List<UwaziEntityInstance>> listSubmittedInstances();
}

File: mobile/src/main/java/rs/readahead/washington/mobile/domain/repository/uwazi/ICollectUWAZITemplatesRepository.kt
package rs.readahead.washington.mobile.domain.repository.uwazi

import kotlinx.coroutines.flow.Flow
import rs.readahead.washington.mobile.domain.entity.uwazi.CollectTemplate
import rs.readahead.washington.mobile.domain.entity.uwazi.ListTemplateResult

interface ICollectUWAZITemplatesRepository {
    suspend fun listBlankTemplates(): Flow<List<CollectTemplate>>
    suspend fun listFavoriteTemplates(): Flow<List<CollectTemplate>>
    suspend fun updateBlankTemplates(listTemplateResult: ListTemplateResult): Flow<ListTemplateResult>
    suspend fun updateBlankTemplatesIfNeeded(listTemplateResult: ListTemplateResult): Flow<ListTemplateResult>
    suspend fun updateBlankTemplate(template: CollectTemplate?): Flow<ListTemplateResult>
}